kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/config/S3ConfigSettings.scala

```diff
@@ -111,11 +111,11 @@ object S3ConfigSettings {
   val PADDING_STRATEGY = s"$CONNECTOR_PREFIX.padding.strategy"
   val PADDING_STRATEGY_DOC =
     "Configure in order to pad the partition and offset on the sink output files. Options are `LeftPad`, `RightPad` or `NoOp`  (does not add padding). Defaults to `LeftPad`."
-  val PADDING_STRATEGY_DEFAULT = "LeftPad"
+  val PADDING_STRATEGY_DEFAULT = ""
 
-  val PADDING_LENGTH         = s"$CONNECTOR_PREFIX.padding.length"
-  val PADDING_LENGTH_DOC     = s"Length to pad the string up to if $PADDING_STRATEGY is set."
-  val PADDING_LENGTH_DEFAULT = 12
+  val PADDING_LENGTH     = s"$CONNECTOR_PREFIX.padding.length"
+  val PADDING_LENGTH_DOC = s"Length to pad the string up to if $PADDING_STRATEGY is set."
+  val PADDING_LENGTH_DEFAULT: Int = -1
```

Comment:
Why is it -1? Did we not agree it will pad by default the file names? Also, -1 is a misleading value. If -1 is accepted, is -199?

Reply:
This is the KCQL default.  This is the default value used in the config def.  If the KCQL is not set then it looks at the KCQL properties.  This happens in the code here:

https://github.com/lensesio/stream-reactor/blob/7d801c5229c69f62062a884957bc88b963ab674e/kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/padding/PaddingService.scala#L39

Agree with your point on -199.  It is not a valid value.  In the code where I'm checking the default I will change it to a < comparison with zero so that it will consider **all** negatives as unset.

Reply:
Making the properties default value any other than -1 (a special value to show "unset") would be incompatible with having a check that indicates if both the properties value and KCQL properties are set and printing a warning.